draft-welzl-iccrg-pacing.md

```diff
@@ -70,11 +70,43 @@ Because of its known advantages, pacing has become common in implementations of
+In this case, an overshoot will occur: packets are forwarded with half their arrival rate, i.e. less than 20 packets can be forwarded during the burst's arrival. The remaining 20 (or more) packets cannot fit into the 10-packet queue. A cwnd of 40 packets is much smaller than the (BDP + queue) limit of 135 packets, and the bottleneck is not fully saturated.
+
+Let us now assume that the flight of 40 packets is instead paced, such that the arrival rate only mildly exceeds the departure rate -- e.g., they arrive at a rate of 60 Mbit/s. When the last packet of this flight arrives at the bottleneck, 5/6 * 39 = 32.5 packets should already have been transferred. Since only complete packets are sent, 32 packets have really been transferred, and the remaining 40-32 = 8 packets fit in the queue. No loss occurs.
+
+This example explains how pacing can enable a rate increase to last longer than without pacing. This makes it more likely that a bottleneck is saturated, such that cwnd reflects the BDP plus the queue length (loss type 2).
+
+### Backing off after the increase
+
+The two loss types explained in {{losstypes}} require a different back-off factor to allow the queue to drain and congestion to dissipate. Specifically, in the single-sender single-bottleneck example above, when a slow start overshoot occurs as loss type 2, a back-off function such as: ssthresh = cwnd * beta with beta >= 0.5 is guaranteed to cause a second loss after the end of loss recovery. This is because, when cwnd exceeds a fully saturated bottleneck (i.e., cwnd > BDP + queue length), cwnd will have grown further by another (BDP + queue length) by the time the sender learns about the loss. In this case, beta = 0.5 will cause ssthresh to exceed (BDP + queue length) again.
+
+Since pacing makes loss type 2 more likely, beta < 0.5 may be a better choice after slow start overshoot when pacing is used.
+
+### Able to work with smaller queues
+
+The probability of loss type 1 in {{losstypes}} is indirectly proportional to the queue length. Pacing therefore enables a rate increase to continue with a smaller queue at the bottleneck than in the case without pacing.
+
+
+## Getting good RTT estimates
+
+Since pacing algorithms generally attempt to spread out packets evenly across an RTT, it is important to have a good RTT estimate. Especially in the beginning of a transfer, when sending the initial window, the only RTT estimate available may be from the SYN-SYN/ACK handshake. Being based on only one sample, this is a very unreliable estimate, and using it to pace the initial window can cause unnecessary delay. This may be the reason why the Linux implementation does not pace the first 10 packets (see {{linux}}).
```

Comment:
Is an RTT estimate also available via a route cache from a recent connection?

Reply:
HOW COULD I FORGET!  This was the whole point I wanted to get at - a plug for RFC 9040  :-)   Will do!